applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -92,6 +121,7 @@ def __CreateResponseFunctions( specified_responses, model ):
         sho_response_functions = ["plane_based_packaging", "mesh_based_packaging", "surface_normal_shape_change"]
         csm_response_functions = ["strain_energy", "mass", "eigenfrequency", "adjoint_local_stress", "adjoint_max_stress"]
         convdiff_response_functions = ["point_temperature"]
+        rans_respone_functions = ["lift_to_drag_response", "average_location_deviation_response"]

```

Comment:
"average_location_deviation_response" doesn't sound very descriptive, but I am also not in the details. Just my first thought when reading this.

Reply:
BTW. the  average_location_deviation_response or however we may call it, would fit into the ShapeOptApp, since it is purely geometrical and not specific to the RANS app.

Reply:
I had a hard time coming up with a name for that. Any suggestions?

Reply:
Just some ideas for names:
- centroid_deviation
- nodes_centroid_deviation
- point_cloud_center_deviation
- ...

I think it is important to highlight, that this is "just" the centroid of the point cloud, but not the real centroid considering the "element masses". 

In any case i would remove the `_response` postfix, also from the `lift_to_drag` to be in line with the existing ones.

Reply:
What is this response supposed to do?

Reply:
@armingeiser It is now moved to shape opt.

@dbaumgaertner This calculates the deviation of the design surface from its initial position. I am using this as a constraint to avoid having rigid body movement when optimizing.

Reply:
So why not changing the name then to exactly what it does: "DeviationToDesignSurface" or "GeometricDeviationToDesignSurface". How the deviation is computed is IMO an implementation detail (even so an important one, I agree).

Reply:
I like the GeometricCentroidDeviation better. It describes clearly that the response value is the deviation from the geoemtric centroid. 

`DeviationToDesignSurface` is nto so clear to me, also since "design_surface" is not "known" in the response, and any model part could be used.

Reply:
I see, makes sense. So let's stick with it.

Reply:
Just one more quick idea @armingeiser: from what you are saying above, what about "DeviationFromGeometricCentroid". ...But your call.

Reply:
This one is good :smile: